marquez/client.py

```diff
@@ -0,0 +1,236 @@
+        path = "datasources/{0}"
+        path_args = [urn]
+        return self.get_request(path, path_args, **kwargs)
+
+    @handle_response
+    def create_datasource(self, name, connection_url, **kwargs):
+        path = "datasources"
+        path_args = []
+        payload = {
+            "name": name,
+            "connectionUrl": connection_url
+        }
+        return self.post_request(path, path_args, payload, **kwargs)
+
+    #  Jobs API
+    @handle_response
+    def create_job(self, job_name, location, input_dataset_urns,
+                   output_dataset_urns, description=None, **kwargs):
+        path = "namespaces/{0}/jobs/{1}"
+        path_args = [self.namespace, job_name]
+        payload = {
+            "inputDatasetUrns": input_dataset_urns,
+            "outputDatasetUrns": output_dataset_urns,
+            "location": location
+        }
+
+        if description:
+            payload['description'] = description
+        return self.put_request(path, path_args, payload, **kwargs)
+
+    @handle_response
+    def get_job(self, job_name, **kwargs):
+        path = "namespaces/{0}/jobs/{1}"
+        path_args = [self.namespace, job_name]
+        if 'timeout' not in kwargs:
+            kwargs['timeout'] = self.timeout
+        return self.get_request(path, path_args, **kwargs)
+
+    @handle_response
+    def get_jobs(self, **kwargs):
+        path = "namespaces/{0}/jobs/"
+        path_args = [self.namespace]
+        return self.get_request(path, path_args, **kwargs)
+
+    @handle_response
+    def create_job_run(self, job_name, job_run_args,
+                       nominal_start_time=None,
+                       nominal_end_time=None, **kwargs):
+        path = "namespaces/{0}/jobs/{1}/runs"
+        path_args = [self.namespace, job_name]
+        payload = {
+            "nominalStartTime": nominal_start_time,
+            "nominalEndTime": nominal_end_time,
+            "runArgs": job_run_args
+        }
+        return self.post_request(path, path_args, payload, **kwargs)
+
+    @handle_response
+    def get_job_run(self, job_run_id, **kwargs):
+        path = "jobs/runs/{0}"
+        path_args = [job_run_id]
+        return self.get_request(path, path_args, **kwargs)
+
+    def mark_job_run_running(self, job_run_id, **kwargs):
+        return self.mark_job_run_as(
+            job_run_id, JobRunState.RUN.value, **kwargs)
+
+    def mark_job_run_completed(self, job_run_id, **kwargs):
+        return self.mark_job_run_as(
+            job_run_id, JobRunState.COMPLETE.value, **kwargs)
+
+    def mark_job_run_failed(self, job_run_id, **kwargs):
+        return self.mark_job_run_as(
+            job_run_id, JobRunState.FAIL.value, **kwargs)
+
+    def mark_job_run_aborted(self, job_run_id, **kwargs):
+        return self.mark_job_run_as(
+            job_run_id, JobRunState.ABORT.value, **kwargs)
+
+    @handle_response
+    def mark_job_run_as(self, job_run_id, status, **kwargs):
```

Comment:
If we have `mark_job_run_as()` defined as a public method, what are the benefits of also defining `mark_job_run_aborted()`, `mark_job_run_failed()`, etc?

Reply:
I should mark it as private.